rdrf/rdrf/templatetags/has_subscribable_email.py

```diff
@@ -0,0 +1,20 @@
+from django import template
+from rdrf.models.definition.models import EmailNotification
+
+register = template.Library()
+
+
+# check whether the registry the user belongs to has subscribable email notifications.
+@register.filter()
+def has_subscribable_email(user):
+    if user.num_registries == 0:
+        return False
+    registries = user.registry.all()
+    for registry in registries:
+        email_notification_data = EmailNotification.objects.filter(registry=registry).values()
```

Comment:
This query & code block could be made more simple, maybe filtering on `subscribable` and using `.exists` like so:

```python
EmailNotification.objects.filter(registry=registry, subscribable=True).exists()
```

Could also try include looking up the user's registry in the same query. Happy to go through subquerying in Django with you, or @ppettitau  can too

Reply:
You can remove the outer loop too and do a single `EmailNotification` lookup to check if `registry` is in the list of `registries` using Django's `__in` ORM operator.
See docs: https://docs.djangoproject.com/en/4.2/ref/models/querysets/#in

This is probably a good one to play around with in the `django-admin shell_plus` console to see what's possible to do with the Django ORM querying.